contributors/devel/sig-architecture/feature-gates.md

```diff
@@ -0,0 +1,64 @@
+# Feature gate progression in Kubernetes
+
+Feature flags are how Kubernetes tracks progression of functionality that is available to a deployer.
+
+## Lifecycle
+
+Feature progress through `Alpha` -> `Beta` -> `GA`. Sometimes we end up deciding that a feature is not going to be supported and we end up marking them as `Deprecated`.
```

Comment:
Might be worth indicating that there is also a final state when the feature gates disappear and the feature became part of the product